### ISSUE TITLE
Add custom bean validator before saving mapped data

### ISSUE THREAD

**Issue by oskarhinc** [benetech/ServiceNet#75]
As the amount of data from partners is going to be really huge, it is required to validate mapped entities before saving them. Otherwise, none of the data will be persisted, because there was a DB exception with a single entity. Doing it manually with each entity is not the best solution, so we should use something like Spring Bean Validation to handle this. It would be nice to have details about beans that were validated with errors stored in the database, so it could be verified and fixed then, but fixing those errors should be a separate feature. 

This should be done after #176

**Comment by wojtekbu** [comment]
@oskarhinc  works fine except that error messages are unformatted. For example 'nullCannot connect with Google Maps API Cannot connect with Google Maps API'

**Comment by oskarhinc** [comment]
Nice natch @wojtekbu . I've improved the reports handling for errors as well now: https://github.com/benetech/ServiceNet/pull/279
